src/views/Player.vue

```diff
@@ -426,33 +445,24 @@ export default {
 
         this.updateJourney("exit-fullscreen")
         this.uploadJson()
-      })
+      });
 
-      var prevTime = -1
-      player.on("timeupdate", async () => {
 
-        // Update watch time if the video is playing
-        if(this.player.playing) {
-          this.watchTime += interval_time;
-        }
-
-        // Record how many times a particular second was visited
-        var currTime = Math.trunc(this.player.currentTime);
-        if (currTime != prevTime) {
-            this.retention[currTime] += 1;
-            prevTime = currTime
-        }
+      // Keep checking when to pop up the question
+      setInterval(() => {
         
         this.plioQuestions.forEach(async (plioQuestion) => {
-
+          
           var question = plioQuestion.item;
           var t = question.time;
+
           if (
-            // "timeupdate" event is called every interval_time millisecond
-            this.player.currentTime > t
-            && this.player.currentTime < t + (interval_time/1000)
-            //&& plioQuestion["state"] == "notshown"
+            // if the seeker is within "popUpPrecisionTime" of the specific question time
+            // then fire this logic
+            this.player.currentTime >= t
+            && this.player.currentTime < t + (popUpPrecisionTime)/1000
           ) {
+            this.player.pause();
             var id = plioQuestion.id
             this.$refs['position' + id.toString()].openModal();
             while (!document.querySelector(".modal")) {
```

Comment:
if `currentTime` is within the range `[t, t+popUpPrecisionTime)`, it will open the modal.

Also moved the line `this.player.pause();` upwards because when it was down the lines, the player didn't pause for a few seconds and kept playing, and that was the reason close questions were being missed